system/ubloxd/pigeond.py

```diff
@@ -260,36 +258,33 @@ def deinitialize_and_exit(pigeon: TTYPigeon | None):
   set_power(False)
   sys.exit(0)
 
-def create_pigeon() -> tuple[TTYPigeon, messaging.PubMaster]:
-  pigeon = None
-
+def init(pigeon: TTYPigeon) -> None:
   # register exit handler
   signal.signal(signal.SIGINT, lambda sig, frame: deinitialize_and_exit(pigeon))
-  pm = messaging.PubMaster(['ubloxRaw'])
 
   # power cycle ublox
   set_power(False)
   time.sleep(0.1)
   set_power(True)
   time.sleep(0.5)
 
-  pigeon = TTYPigeon()
-  return pigeon, pm
+  init_baudrate(pigeon)
+  init_pigeon(pigeon)
+
+def run_receiving(duration: int = 0):
+  pm = messaging.PubMaster(['ubloxRaw'])
 
-def run_receiving(pigeon: TTYPigeon, pm: messaging.PubMaster, duration: int = 0):
+  pigeon = TTYPigeon()
+  init(pigeon)
 
   start_time = time.monotonic()
-  def end_condition():
-    return True if duration == 0 else time.monotonic() - start_time < duration
-
   last_almanac_save = time.monotonic()
-  while end_condition():
+  while (duration == 0) or (time.monotonic() - start_time < duration):
     dat = pigeon.receive()
     if len(dat) > 0:
       if dat[0] == 0x00:
```

Comment:
@robbederks you remember why this is here? I don't see why this on its own would be indicative of corrupt data (e.g. what enforces that you're getting the start of a ublox packet?)